src/sentry/workflow_engine/handlers/condition/event_frequency_base_handler.py

```diff
@@ -25,9 +32,14 @@ class _QSTypedDict(TypedDict):
 class BaseEventFrequencyConditionHandler(ABC):
     @property
     @abstractmethod
-    def intervals(self) -> dict[str, tuple[str, timedelta]]:
+    def base_handler(self) -> Self:
+        # frequency and percent conditions can share the same base handler to query Snuba
         raise NotImplementedError
```

Comment:
added this to prepare for delayed processing. all `EventUniqueUserFrequency` conditions make Snuba queries in the same way, percent conditions just make 1 extra query to compare across some interval (e.g. 1 week)